pkg/apis/core/types.go

```diff
@@ -3853,12 +3859,16 @@ type ServiceSpec struct {
 	// +optional
 	LoadBalancerSourceRanges []string
 
-	// externalTrafficPolicy denotes if this Service desires to route external
-	// traffic to node-local or cluster-wide endpoints. "Local" preserves the
-	// client source IP and avoids a second hop for LoadBalancer and Nodeport
-	// type services, but risks potentially imbalanced traffic spreading.
-	// "Cluster" obscures the client source IP and may cause a second hop to
-	// another node, but should have good overall load-spreading.
+	// externalTrafficPolicy describes the endpoint-selection policy for traffic
+	// received from outside the cluster on one of the Service's external IPs/ports
+	// (NodePorts, ExternalIPs, and LoadBalancer IPs). "Cluster" (the default) uses
+	// the standard behavior of routing to all endpoints (possibly modified by
+	// topology and other features), which should have good overall load-spreading but
+	// normally obscures the client source IP. "Local" delivers traffic only to
+	// node-local endpoints, avoiding a second hop to another node for traffic that
+	// was already load-balanced externally, and preserving the client source IP. Note
+	// that externalTrafficPolicy may not apply to traffic sent to an external IP/port
+	// from a client inside the cluster.
```

Comment:
Should we detail the sepcial cases here?

Reply:
Is the API documentation really the right place to dive into the semantics of "pod to same-node NodePort" vs "pod to remote NodePort", or AWS-style load balancers vs GCP-style load balancers?

(Is the difference in behavior between pod-to-same-node-NodePort and pod-to-remote-NodePort even canonically part of the definition of Local traffic policy? Or is that just the behavior of iptables kube-proxy, where we care about preserving the specific backward-compatible behavior _in iptables kube-proxy_ but don't necessarily care if other impementations behave precisely the same way? Considering, for example, that in some plugins with a more "centrally-programmed" data plane (like ovn-kubernetes), it's not only _possible_, but actually _easier_ to make pod-to-NodePort always behave the same way regardless of whether it's same-node or different-node.)

I guess the API docs should describe the behavior that is intentional and reliable and encouraged... so I'd say that we don't say anything about pod/node to NodePort, and that for pod/node to externalIP/LBIP it always has Cluster semantics.

Reply:
I can buy the "describe the behavior that is intentional" theme.  We already have implementation-defined things like localhost and we don't call that out, either.  Pod->nodeport is a weird behavior anyway. Your approach SGTM